spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -385,6 +389,9 @@ private static void mergeIcebergHadoopConfs(
       }
     }
 
+    ValidationException.check(tasks.stream().noneMatch(TableScanUtil::hasDeletes),
```

Comment:
I'm a little confused by this check, shouldn't we be using `tasks()` when planning deletes with the non-batch reader?

Reply:
Yes, I think we should use `tasks()`

Reply:
I've fixed this. This should also have been in `planBatchInputPartitions` because the check is intended to ensure that we don't attempt to read using batch when there are deletes to apply. This was here because I initially disabled all reads with deletes to apply before I implemented deletes for row-based reads.

Reply:
Sgtm :) I was just really confused how anything worked at all with this check, I assumed you had some trick up your sleeve  I just didn't get.

Reply:
You really shouldn't trust me that much.